x/programs/runtime/call_context.go

```diff
@@ -0,0 +1,88 @@
+// Copyright (C) 2023, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package runtime
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"reflect"
+
+	"github.com/ava-labs/avalanchego/ids"
+
+	"github.com/ava-labs/hypersdk/codec"
+)
+
+var errCannotOverwrite = errors.New("trying to overwrite set field")
+
+type CallContext struct {
+	r               *WasmRuntime
+	defaultCallInfo CallInfo
+}
+
+func (c CallContext) createCallInfo(callInfo *CallInfo) (*CallInfo, error) {
+	newCallInfo := *callInfo
+	resultInfo := reflect.ValueOf(&newCallInfo)
+	callInfoTypeInfo := reflect.TypeOf(CallInfo{})
+	defaults := reflect.ValueOf(c.defaultCallInfo)
```

Comment:
What is the reason why you create a new `callInfoTypeInfo` here, can't we use `defaults` ?